### ISSUE TITLE
Compile Problems with ROCT-Thunk-Interface branch ("fxkamd/drm-next-wip") and 1.8.x branch ROCR Runtime

### ISSUE THREAD

**Issue by saitam757** [RadeonOpenCompute/ROCT-Thunk-Interface#22]
I tried to get ROCT and ROCR work on a system with mainline Kernels (4.18 on Manjaro, 4.17 on Solus) and faced some probelms. When compiling ROCR Runtime I got the error that _HSA_ENGINE_VERSION uCodeEngineVersions_ in _hsakmttypes.h_ was not defined. I did some ugly merges of _hsakmttypes.h_ and _topology.c_ with master of ROCT-Thunk-Interface to get this typedef and finally I got ROCR compiled. But get a segmentation fault when executing _vector_copy_ sample. Seems to be that ROCT and ROCR became incompatible ?

**Comment by fxkamd** [comment]
ROCr is depending on a ROCt API change that's missing in my WIP branch (Export microcode version of sDMA). I cherry-picked the API change to the branch now. Please give it a try.

**Comment by saitam757** [comment]
Thank you for your fast replay. Short report what happened during compiling of ROCt (gcc 8.1.1, Manjaro, Kernel 4.18.0-1, RX-580):
`topology.c:860:3: Error: »strncpy« specified limit 64 is exactly the size of the target [-Werror=stringop-truncation] strncpy((char *)props->AMDName, hsa_gfxip->amd_name, sizeof(props->AMDName))`
Solved it by adding _-Wno-stringop-truncation_ to _CmakeList.txt_ . Not the nicest solution but it compiled afterwards. When compiling ROCr1.8.x got the error
`ROCR-Runtime/src/core/runtime/runtime.cpp:260:71: Error: »void* memset(void*, int, size_t)«`
Corrected the line with
`memset(static_cast<void*>(&link_matrix_[0]), 0,
         link_matrix_.size() * sizeof(hsa_amd_memory_pool_link_info_t));`
and it compiled. I installed the _HSA-Runtime-AMD_ libraries and did a test with vector_copy sample program. Result is
`Initializing the hsa runtime succeeded.
Checking finalizer 1.0 extension support succeeded.
Generating function table for finalizer succeeded.
Getting a gpu agent succeeded.
Querying the agent name succeeded.
The agent name is gfx803.
Querying the agent maximum queue size succeeded.
The maximum queue size is 131072.
Creating the queue succeeded.
"Obtaining machine model" succeeded.
"Getting agent profile" succeeded.
` **Segmentation fault (core dumped)**
Maybe this could be interesting to you for debugging.
